src/git.ts

```diff
@@ -0,0 +1,612 @@
+        return;
+    }
+    if (!vscode.workspace.getConfiguration('neuropilot').get('permission.gitOperations', false) && !vscode.workspace.getConfiguration('neuropilot').get('permission.gitRemotes')) {
+        NEURO.client?.sendActionResult(actionData.id, true, 'You do not have permission to perform Git operations.');
+        return;
+    }
+
+    const repo = git.repositories[0];
+
+    NEURO.client?.sendActionResult(actionData.id, true)
+
+    repo.pull().then(() => {
+        NEURO.client?.sendContext(`Pulled commits from remote.`);
+    }
+    , (err: string) => {
+        NEURO.client?.sendContext(`Failed to pull commits from remote`);
+        logOutput("ERROR", `Failed to pull commits: ${err}`)
+    });
+}
+
+export function handlePushGitCommits(actionData: any) {
+    if (!git) {
+        NEURO.client?.sendActionResult(actionData.id, true, 'Git extension not available.');
+        return;
+    }
+    if (!vscode.workspace.getConfiguration('neuropilot').get('permission.gitOperations', false) && !vscode.workspace.getConfiguration('neuropilot').get('permission.gitRemotes')) {
+        NEURO.client?.sendActionResult(actionData.id, true, 'You do not have permission to perform Git operations.');
+        return;
+    }
+
+    const repo = git.repositories[0];
+    const remoteName: string = actionData.params?.remoteName;
+    const branchName: string = actionData.params?.branchName;
+    const forcePush: boolean = actionData.params?.forcePush || false;
+
+    NEURO.client?.sendActionResult(actionData.id, true)
+
+    repo.push(remoteName, branchName, forcePush).then(() => {
+        NEURO.client?.sendContext(`Pushed commits${remoteName && ` to remote "${remoteName}"`}${branchName && `, branch "${branchName}"`}.${forcePush === true && " (forced push)"}`);
+    }, (err: string) => {
+        NEURO.client?.sendContext(`Failed to push commits to remote "${remoteName}"`);
+        logOutput("ERROR", `Failed to push commits: ${err}`)
+    });
+}
+
+/**
+ * THESE ACTIONS ARE CONSIDERED DANGEROUS REMOTE OPERATIONS
+ * Requires neuropilot.permission.editRemoteData to be enabled, IN ADDITION to neuropilot.permission.gitRemotes.
+ */
+
+export function handleAddGitRemote(actionData: any) {
+    if (!git) {
+        NEURO.client?.sendActionResult(actionData.id, true, 'Git extension not available.');
+        return;
+    }
+    if (!vscode.workspace.getConfiguration('neuropilot').get('permission.gitOperations', false) && (!vscode.workspace.getConfiguration('neuropilot').get('permission.gitRemotes', false) && (!vscode.workspace.getConfiguration('neuropilot').get('permission.editRemoteData', false)))) {
+        NEURO.client?.sendActionResult(actionData.id, true, 'You do not have permission to perform Git operations.');
+        return;
+    }
+
+    const repo = git.repositories[0];
+    const remoteName: string = actionData.params?.remoteName;
+    const remoteUrl: string = actionData.params?.remoteUrl;
```

Comment:
Property name in the schema is `remoteURL`.
```suggestion
    const remoteUrl: string = actionData.params?.remoteURL;
```